tornjak-frontend/src/components/tornjak-helper.js

```diff
@@ -119,19 +110,29 @@ class TornjakHelper extends Component {
 
   // workloadEntry takes in an entry metadata and workload attestor info for specified agents
   // returns entry metadata info for dashboard table
-  workloadEntry(entry, workLoadAttestorInfo) {
+  workloadEntry(entry, workLoadAttestorInfo, globalAgents, globalEntries) {
     var thisSpiffeId = this.SpiffeHelper.getEntrySpiffeid(entry)
     var thisParentId = this.SpiffeHelper.getEntryParentid(entry)
+    var canonicalParentId = thisParentId
     // get tornjak metadata
-    var metadata_entry = this.SpiffeHelper.getAgentMetadata(thisParentId, workLoadAttestorInfo);
+    var parentAgent = globalAgents.find(a => this.SpiffeHelper.getAgentSpiffeid(a) === thisParentId)
+    if (parentAgent === undefined) {
+        var agent = globalEntries.find(e => this.SpiffeHelper.getEntrySpiffeid(e) === thisParentId)
+        if (agent !== undefined) {
+            canonicalParentId = this.SpiffeHelper.getCanonicalAgentSpiffeid(agent, globalAgents)
+        }
+    }
+
     var plugin = "None"
     var cluster = "None"
-    if (metadata_entry["plugin"].length !== 0) {
-      plugin = metadata_entry["plugin"]
+    var metadataEntry = this.SpiffeHelper.getAgentMetadata(canonicalParentId, workLoadAttestorInfo);
+    if (metadataEntry["plugin"].length !== 0) {
+      plugin = metadataEntry["plugin"]
     }
-    if (metadata_entry["cluster"].length !== 0) {
-      cluster = metadata_entry["cluster"]
+    if (metadataEntry["cluster"].length !== 0) {
```

Comment:
alternative:
```
if (metadataEntry["cluster"]?.length) {
```

Reply:
File not of concern on this PR. Will be addressed in the next consecutive PR.